### ISSUE TITLE
lab1#NON-CRITICAL# - Some things to think about connected to TASK 1 

### ISSUE THREAD

**Issue by ArthurAlexandersson** [Chimbobedeh/TDA553-lab1#1]
## NON-CRITICAL ##
- Can I create a new car with trim without duplicate code?
- Can I create a new car with turbo without duplicate code?
- Can I create a new car with both trim and turbo without duplicate code?

**Comment by rilind404** [comment]
Currently no. We have however received ideas on how to fix this from a recent lecture with Robin and are planning to fix this.

The idea is to create more subclasses to `Car` for example `Engine`, `VehicleBody` and then create subclasses to `EngineTurbo` and `VehicleBodyTrim` respectively. Then we can have cars delegate turbo, trim (turbo & trim) functionality to these classes.

We are currently working with lab 2 and will get on with implementing these ideas soon.

**Comment by ArthurAlexandersson** [comment]
Very good.

**Comment by rilind404** [comment]
We have created following classes:

-Engine
-CarBody
-EngineTurbo
-BodyTrim

EngineTurbo is a subclass to Engine, BodyTrim is subclass to CarBody.
Car should* delegate to Engine and CarBody when a vehicle is created. Then, as an example, Ferrari will delegate to EngineTurbo and BodyTrim to get both of these functionalities.

*I say should as we have created the classfiles but now quite implemented them. We found this task to be easier said than done and preferred to finish the required tasks for lab2. We spoke with TA and this implementation is preferred, but not required hence the choice to leave them unimplemented.
